2_process.R

```diff
@@ -51,13 +51,11 @@ p2_targets_list <- list(
   # Match PRMS stream segments to observation site ids
   tar_target(
     p2_sites_w_segs,
-    get_site_flowlines(
-      p1_reaches_sf,
-      p2_site_list,
-      sites_crs = 4269,
-      max_matches = 1,
-      search_radius = 0.1
-    )
+    get_site_flowlines(p1_reaches_sf, p2_site_list, sites_crs = 4269,
+                       max_matches = 1, search_radius = 1000) %>%
+      # only retain sites that get matched to flowlines based on a 
+      # cutoff value of bird_dist_to_subseg_m:
+      filter(bird_dist_to_subseg_m <= 500)
```

Comment:
Yeah, good question. They're redundant and I recognize that it's also kind of confusing that I've set them at different values here. If we just set the `search_radius` we'll still return _every_ DO site regardless of whether it snapped to a flowline (because we join the sparser flowline indices table to the original site table within `get_site_flowlines())`. So the output looks like:

```
> p2_sites_w_segs %>% as.data.frame %>% head
                     site_id                                      site_name count_days_nwis count_days_discrete count_days_total       lon
1 11NPSWRD_WQX-HOFU_BOYER_01            FRENCH CREEK AT PA ROUTE 345 BRIDGE              NA                   1                1 -75.76927
2 11NPSWRD_WQX-HOFU_BOYER_02                PINE CREEK AT HARMONYVILLE ROAD              NA                   1                1 -75.77396
3 11NPSWRD_WQX-HOFU_BOYER_03 S BRANCH FRENCH CREEK AT PA ROUTES 345 AND 401              NA                   1                1 -75.79341
4 11NPSWRD_WQX-HOFU_BOYER_04       FRENCH CREEK AT COUNTY PARK ROAD (T-452)              NA                   1                1 -75.72541
5 11NPSWRD_WQX-HOFU_BOYER_05         S BRANCH FRENCH CK AT IRON BRIDGE ROAD              NA                   1                1 -75.71382
6 11NPSWRD_WQX-HOFU_BOYER_06      FRENCH CREEK AT SHEEDER MILL ROAD (T-491)              NA                   1                1 -75.62782
       lat datum       org_id   data_src_combined subsegid bird_dist_to_subseg_m segidnat
1 40.20343 NAD83 11NPSWRD_WQX Harmonized_WQP_data    863_1              2.249200     2297
2 40.18785 NAD83 11NPSWRD_WQX Harmonized_WQP_data     <NA>                    NA       NA
3 40.14793 NAD83 11NPSWRD_WQX Harmonized_WQP_data    575_1            929.729381     2009
4 40.16987 NAD83 11NPSWRD_WQX Harmonized_WQP_data    863_1             13.987362     2297
5 40.15501 NAD83 11NPSWRD_WQX Harmonized_WQP_data     <NA>                    NA       NA
6 40.15085 NAD83 11NPSWRD_WQX Harmonized_WQP_data    863_1              7.507442     2297
```

My question was whether we'd prefer to return a sparser table that omits sites like those above where `segidnat == NA` because the sites didn't match within the given `search_radius`. Omitting the NA sites would make the output of this target cleaner, and potentially(?) speed up a couple downstream steps, like when we join this site table to the met data table.

Reply:
I added the additional filter step because it seemed like I was missing a couple sites compared to when I set `search_radius = 500`.

```
> p2_sites_w_segs <- get_site_flowlines(p1_reaches_sf, p2_site_list, sites_crs = 4269,
+                                       max_matches = 1, search_radius = 1000) %>%
+     filter(bird_dist_to_subseg_m <= 500)
matching flowlines with reaches...
rejoining with other geometries
> dim(p2_sites_w_segs)
[1] 946  13
> 
> p2_sites_w_segs <- get_site_flowlines(p1_reaches_sf, p2_site_list, sites_crs = 4269,
+                                       max_matches = 1, search_radius = 500)
matching flowlines with reaches...
rejoining with other geometries
> dim(p2_sites_w_segs)
[1] 944  13
```

The two sites that get omitted from the latter approach do seem to be within 500 m of `p1_reaches_sf`. It's probably not worth worrying about these particular sites too much, but I don't understand why they're not included in the output if we just set `search_radius = 500` (is there some quirk with the k-nearest neighbors approach employed by `nhdplusTools::get_flowline_index`?). 
```
              site_id              site_name count_days_nwis count_days_discrete count_days_total       lon      lat datum       org_id
1 31DELRBC_WQX-091002 Reedy Island (RM 54.9)              NA                 116              116 -75.55344 39.51278 NAD83 31DELRBC_WQX
2     31DELRBC-091002 Reedy Island (RM 54.9)              NA                  70               70 -75.55333 39.51278 NAD83     31DELRBC
    data_src_combined subsegid bird_dist_to_subseg_m segidnat
1 Harmonized_WQP_data   2759_1              241.0485     4193
2 Harmonized_WQP_data   2759_1              231.8473     4193

```

Reply:
@jsadler2 I tweaked this function a bit to get around any edge effects in the radius used by `nhdplusTools::get_flowline_index` (ultimately from `RANN::nn2` radius search) and to simplify the target definition. The function to build target `p2_sites_w_segs` now returns a "sparse" list of observation sites that only includes sites that are matched to a segment within the specified `search_radius`.